Amplify/Categories/API/AuthProvider/APIAuthProviderFactory.swift

```diff
@@ -18,11 +18,19 @@ open class APIAuthProviderFactory {
     open func oidcAuthProvider() -> AmplifyOIDCAuthProvider? {
         return nil
     }
+
+    open func functionAuthProvider() -> AmplifyFunctionAuthProvider? {
+        return nil
+    }
```

Comment:
we choose the name `function` to avoid leaking service implementation detail into the library public API